### ISSUE TITLE
How can I build a model using the Functional APIs?

### ISSUE THREAD

**Issue by italodamato** [tensorflow/recommenders#163]
I'm trying to build a model using the functional APIs to define a `keras.Model` object instead of creating a subclass like in the tutorials. However, I'm struggling to understand how to tie everything together (e.g. compute the loss, pass the right variables in the Input layers).

Before proceeding further I just want to know if this is even possible/advised? The reason why I'm doing this is that I need to share embedding layers across features.

Any advice is really appreciated.

**Comment by maciejkula** [comment]
We recommend using subclassed models: they are a lot more flexible and easy to understand.

You can share embedding variables between features by reusing the same instance of an `Embedding` layer with different features inside your model.

For example the following snippet will reuse the same embedding variables for both features:
```python
embeddings = Embedding(...)

_ = embeddings(feature_a)
_ = embeddings(feature_b)
```

**Comment by italodamato** [comment]
Thank you, it makes sense. In my understanding, I thought the Functional API was considered more flexible than the subclassed models. (I'm a big fan!)
